### ISSUE TITLE
Fix obfuscation for DB on the main thread perf issues

### ISSUE THREAD

**Issue by davidrothera** [getsentry/sentry#52851]
### Platform

iOS

### Installed

Swift Package Manager

### Version

8.8.0

### Steps to Reproduce

* Enable CoreData tracing
* Perform DB action on the main thread (to trigger a performance issue)

### Expected Result

* Look at performance trace created by performing DB work on the main thread (e.g. with SwiftUI) and see where the call originated from in my app

### Actual Result

* All calls in the call-stack are redacted; both system frameworks as well as in-app ones (my main app bundle)

**Comment by philipphofmann** [comment]
The `FileIOMainThreadDetector` has `_deobfuscate_module` and `_deobfuscate_function`, which are only called in `_fingerprint`

https://github.com/getsentry/sentry/blob/36269235a413746346e56a38682dbeaa27ef8dc2/src/sentry/utils/performance_issues/detectors/io_main_thread_detector.py#L162C9-L176

The `DBMainThreadDetector` doesn't have `_deobfuscate_module` and `_deobfuscate_function`. So my idea is to move `_deobfuscate_module`, `_deobfuscate_function`, and `_prepare_deobfuscation` to `BaseIOMainThreadDetector`, but the `_fingerprint` method for `DBMainThreadDetector` looks a bit different than the one of FileIOMainThreadDetector: 

https://github.com/getsentry/sentry/blob/36269235a413746346e56a38682dbeaa27ef8dc2/src/sentry/utils/performance_issues/detectors/io_main_thread_detector.py#L214-L223

I'm unsure what happens when I change the fingerprint. @wmak, can you help me to understand what changes are required to fix this issue?

**Comment by wmak** [comment]
Actually so that deobfuscation is only for determining the fingerprint. How I see it is there's two choices here

- Deobfuscate on the frontend/UI before we present it to the user
- Deobfuscate on ingestion so we only do it once
  - I believe both View Hierarchies and Profiles does this so we avoid repeated deobfuscation calls
  - cc @narsaynorath I believe you know more about how we do it for View Hierarchies

**Comment by narsaynorath** [comment]
Right, View Hierarchies is processed on ingestion so we only store the deobfuscated version. We can find that code here:
https://github.com/getsentry/sentry/blob/master/src/sentry/lang/java/utils.py#L97C34-L97C42

**Comment by shruthilayaj** [comment]
I looked into this today, here are my thoughts/theories:
1. `<redacted>` is the initial value set by the sentry cocoa SDK when we [initialize the frame](https://github.com/getsentry/sentry-cocoa/blob/main/Sources/Sentry/SentryFrame.m#L11) - so I think that's what is being sent from the SDK and doesn't have anything to do with the de-obfuscation we do during performance issue detection @philipphofmann - can you fact check me? If this is the case - is there something going on in the SDK where "function" is not being set properly on SentryFrame?
2. Going through the performance issue detection code, we don't actually mutate the event object after we de-obfuscate the function, so there's nothing we're doing there that would result in these functions being populated vs not - I've also not been able to replicate this issue with our demo app - ![Image](https://github.com/user-attachments/assets/4424038c-afc4-43b3-adaf-40276325531a)

**Comment by philipphofmann** [comment]
It doesn't make sense that the Cocoa SDK sets the initial function value to <redacted>. I created an issue to fix this in the next major https://github.com/getsentry/sentry-cocoa/issues/4686.

@shruthilayaj, for app store builds, the SDK can't set the function name, and it will be `<redacted>` for all frames usually. So there is some symbolication happening, otherwise, all frames would be `<redacted>`. The one redacted in the callstack above is caused by https://github.com/getsentry/sentry-cocoa/issues/1547 and is something you can ignore for now. So I think this issue is already fixed, cause otherwise all frames for the stacktrace you posted should be `<redacted>`.

**Comment by shruthilayaj** [comment]
> for app store builds, the SDK can't set the function name, and it will be <redacted> for all frames usually. So there is some symbolication happening, otherwise, all frames would be <redacted>

ah, I see thanks for clarifying - I was unsure if something else was expected out of the SDK with function names. 
@philipphofmann and just for completeness what does `<redacted>` mean when we know symbolication is happening? Are we missing debug information to unscramble those filenames?

**Comment by philipphofmann** [comment]
> just for completeness what does <redacted> mean when we know symbolication is happening? Are we missing debug information to unscramble those filenames?

Yes, either we have some system lib dSYMs missing or the user didn't upload them.

**Comment by shruthilayaj** [comment]
Okay, I'm closing this as it looks like we have expected behaviour then.
